### DISCUSSION TITLE
How logits is extracted?

### DISCUSSION THREAD

**Discussion by DavidZyy** [ggml-org/llama.cpp#14375] [Q&A]
At [this line](https://github.com/ggml-org/llama.cpp/blob/73e53dc834c0a2336cd104473af6897197b96277/src/llama-context.cpp#L1068), logits is copied from t_logits to logits_out, with `offset = 0`, `size = n_outputs*n_vocab*sizeof(float)`. However, it seems that logits of output is not always contiguous at tensor memory, for example the picture at below:

![image](https://github.com/user-attachments/assets/9c1f7ff2-287d-40c1-83be-04ec6c4335cf)

So how does this case is handled?
